datasource/virtual_machine/filters.go

```diff
@@ -0,0 +1,132 @@
+package virtual_machine
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+// Filter machines by matching their names against defined regular expression.
+func filterByNameRegex(vmList []*object.VirtualMachine, nameRegex string) []*object.VirtualMachine {
+	re, _ := regexp.Compile(nameRegex)
+	result := make([]*object.VirtualMachine, 0)
+	for _, i := range vmList {
+		if re.MatchString(i.Name()) {
+			result = append(result, i)
+		}
+	}
+	return result
+}
+
+// Filter machines by template attribute. Only templates will pass the filter.
+func filterByTemplate(driver *VCenterDriver, vmList []*object.VirtualMachine) ([]*object.VirtualMachine, error) {
+	result := make([]*object.VirtualMachine, 0)
+	for _, i := range vmList {
+		isTemplate, err := i.IsTemplate(driver.ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "error checking if VM is a tempalte")
+		}
+
+		if isTemplate {
+			result = append(result, i)
+		}
+	}
+	return result, nil
+}
+
+// Filter machines by node placement. Only machines that are stored on the defined node will pass the filter.
```

Comment:
```suggestion
// filterByNode filters machines by node placement. Only machines that are stored on the defined node will pass the filter.
```

Though, as mentioned in an earlier comment, the use of `host` instead of `node` is preferred.